app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt

```diff
@@ -117,4 +125,20 @@ object TestHelper {
             }
         }
     }
+
+    fun setNetworkConnection(status: String) {
+        val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+        when (status) {
+            "ON" -> {
+                mDevice.executeShellCommand("svc data enable")
```

Comment:
Perhaps we should assert that the network is enabled/disabled using the app APIs (`ConnectivityManager` or whatever it is) after running these commands. I'm concerned about a few cases:
- it's also possible to transfer data over ADB, which the firebase data centers may be doing for stability reasons
- we may not be able to execute these commands on certain devices, e.g. firebase may disable shell access for security reasons
- it may take some time for this command to execute so the test isn't actually testing what we want it to test. For example, after enabling wifi, it takes a few seconds before the user actually has network connectivity again.

Reply:
Thanks for pointing these out. 

> Perhaps we should assert that the network is enabled/disabled using the app APIs (`ConnectivityManager` or whatever it is) after running these commands. 
> * it may take some time for this command to execute so the test isn't actually testing what we want it to test. For example, after enabling wifi, it takes a few seconds before the user actually has network connectivity again.

Working on this now.

> * we may not be able to execute these commands on certain devices, e.g. firebase may disable shell access for security reasons

What I've found so far is this response from a Firebase engineer, on[ Firebase community](https://firebase-community.slack.com/archives/C1MTSQ5QT/p1611169571083500): 
" williamhester  1 month ago
We don't support adb shell commands, but your Instrumentation test may run shell commands (we can't really stop you)."

Reply:
@mcomella Took your suggestions and made some changes: added Idling resources to wait to be online/offline and fixed some nits, if you care to have another look. Thanks!